tests/phpunit/FunctionalRowTest.php

```diff
@@ -0,0 +1,123 @@
+<?php
+
+namespace Keboola\DbWriter\Snowflake\Tests;
+
+use Keboola\Csv\CsvFile;
+use Keboola\DbWriter\Snowflake\Test\S3Loader;
+use Keboola\DbWriter\Writer\Snowflake;
+use Keboola\StorageApi\Client;
+use Symfony\Component\Process\Process;
+
+class FunctionalRowTest extends BaseTest
+{
+    private const PROCESS_TIMEOUT_SECONDS = 180;
+
+    protected $dataDir = __DIR__ . '/../data/functional-row';
+
+    /** @var Snowflake */
+    private $writer;
+
+    private $config;
+
+    protected $tmpRunDir;
+
+    public function setUp(): void
+    {
+        // cleanup & init
+        $this->tmpRunDir = '/tmp/' . uniqid('wr-db-snowflake_row_');
+        mkdir($this->tmpRunDir . '/in/tables/', 0777, true);
+        $this->config = $this->initConfig();
+
+        $writer = $this->getWriter($this->config['parameters']);
+        if ($writer instanceof Snowflake) {
+            $this->writer = $writer;
+        } else {
+            $this->fail('Writer factory must init Snowflake Writer');
+        }
+
+        $writer = $this->getWriter($this->config['parameters']);
+        $s3Loader = new S3Loader(
+            $this->dataDir,
+            new Client([
+                'token' => getenv('STORAGE_API_TOKEN'),
+            ])
+        );
+
+        // clean destination DB
+        $writer->drop($this->config['parameters']['dbName']);
+
+        // upload source files to S3 - mimic functionality of docker-runner
+        $srcManifestPath = $this->dataDir . '/in/tables/' . $this->config['parameters']['tableId'] . '.csv.manifest';
+        $manifestData = json_decode((string) file_get_contents($srcManifestPath), true);
+        $manifestData['s3'] = $s3Loader->upload($this->config['parameters']['tableId']);
+
+        $dstManifestPath = $this->tmpRunDir . '/in/tables/' . $this->config['parameters']['tableId'] . '.csv.manifest';
+        file_put_contents(
+            $dstManifestPath,
+            json_encode($manifestData)
+        );
+    }
+
+    public function testRun()
```

Comment:
please also add a `testTestConnection` for configRow

Reply:
added